dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/BulkPatchSharingTest.java

```diff
@@ -0,0 +1,245 @@
+
+        String payload = IOUtils.toString( new ClassPathResource( "patch/bulk_sharing_patch.json" ).getInputStream(),
+            StandardCharsets.UTF_8 );
+        assertStatus( HttpStatus.OK, PATCH( "/dataElements/sharing", payload ) );
+
+        JsonIdentifiableObject saveDeA = GET( "/dataElements/{uid}", deAId ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+        JsonIdentifiableObject saveDeB = GET( "/dataElements/{uid}", deBId ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+        assertEquals( 2, saveDeA.getSharing().getUsers().size() );
+        assertEquals( 2, saveDeB.getSharing().getUsers().size() );
+    }
+
+    /**
+     * Payload contains two DataElements but only one exists in database.
+     *
+     * @throws IOException
+     */
+    @Test
+    public void testApplyPatchWithInvalidUid()
+        throws IOException
+    {
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userA", userAId ) ) );
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userB", userBId ) ) );
+        userCId = assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userC" ) ) );
+
+        assertStatus( HttpStatus.CREATED,
+            POST( "/dataElements", toJsonString( createDataElement( 'B', deBId, userCId ) ) ) );
+
+        HttpResponse response = PATCH( "/dataElements/sharing", "patch/bulk_sharing_patch.json" );
+        assertEquals( HttpStatus.CONFLICT, response.status() );
+        assertEquals( "Invalid UID `" + deAId + "` for property `DataElement`", getFirstErrorMessage( response ) );
+
+        JsonIdentifiableObject savedDeB = GET( "/dataElements/{uid}", deBId ).content( HttpStatus.OK )
+            .as( JsonIdentifiableObject.class );
+        assertEquals( 2, savedDeB.getSharing().getUsers().size() );
+        assertNotNull( savedDeB.getSharing().getUsers().get( userAId ) );
+        assertNotNull( savedDeB.getSharing().getUsers().get( userBId ) );
+    }
+
+    /**
+     * Payload contains two DataElements but only one exists in database. Atomic
+     * = true so no sharing settings will be saved.
+     *
+     * @throws IOException
+     */
+    @Test
+    public void testApplyPatchWithInvalidUidAtomic()
+        throws IOException
+    {
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userA", userAId ) ) );
+        assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userB", userBId ) ) );
+        userCId = assertStatus( HttpStatus.CREATED, POST( "/users", createUser( "userC" ) ) );
+
+        assertStatus( HttpStatus.CREATED,
+            POST( "/dataElements", jsonMapper.writeValueAsString( createDataElement( 'B', deBId, userCId ) ) ) );
+
+        String payload = IOUtils.toString( new ClassPathResource( "patch/bulk_sharing_patch.json" ).getInputStream(),
+            StandardCharsets.UTF_8 );
+        HttpResponse response = PATCH( "/dataElements/sharing?atomic=true", payload );
```

Comment:
Could also use filename instead of `payload`

Reply:
fixed